### ISSUE TITLE
plots.plot_poisson_consistency_test is not capturing inf statistic

### ISSUE THREAD

**Issue by pabloitu** [SCECcode/pycsep#45]
For an `EvaluationResult` object with attribute `observed_statistic` which equals to inf, (such as the case in the S-test of a forecast with bin rates of 0.0) the plotting function does not capture correctly this issue, but rather lets `ax.set_xlim()` throw the error. 

https://github.com/SCECcode/pycsep/blob/22e964021c73bf33fdd3567bbcd8f6c2b04ef7a4/csep/utils/plots.py#L1051

To reproduce
#### input
```
import csep
import numpy
result_dict = {'name': 'Poisson S-Test',
               'sim_name': 'Dummy model',
               'obs_name': None,
               'obs_catalog_repr': '',
               'quantile': 0.0,
               'observed_statistic': -numpy.inf,
               'test_distribution': [-150., -145.],
               'status': 'normal',
               'min_mw': 4.95,
               'type': 'EvaluationResult'}
Result = csep.core.poisson_evaluations.EvaluationResult.from_dict(result_dict)
csep.utils.plots.plot_poisson_consistency_test(Result)
```
#### output
```
  File ".../pycsep/venv/lib/python3.7/site-packages/matplotlib/axes/_base.py", line 3213, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
```

Do you think we can make a try/except blockthat indicates that this model is diverging, rather than let matplotlib throws the error? Or plot something for this model that indicates that is diverging...  for example something like:

![image](https://user-images.githubusercontent.com/69042881/94690579-96e78180-0330-11eb-8dd8-58180279960c.png)

Let me know

**Comment by wsavran** [comment]
im not sure what circumstances would cause this issue except for a forecast that prescribes a zero rate in some cell. regardless, maybe we address this. do you wanna try and make a fix? if not, ill include it in one of the next prs

**Comment by pabloitu** [comment]
Perfect. I'll make all the consistency tests  of my models, and then do the fix, so I don't mess with other types of tests.

**Comment by wsavran** [comment]
sounds good. since this issue impacts only the plotting, this fix can be made at any time.

**Comment by pabloitu** [comment]
This is solved in #57
